homeassistant/components/hausbus/device.py

```diff
@@ -0,0 +1,158 @@
+"""Representation of a Haus-Bus device."""
+from pyhausbus.de.hausbus.homeassistant.proxy.controller.params.EFirmwareId import (
+    EFirmwareId,
+)
+
+from homeassistant.helpers.device_registry import DeviceInfo
+
+from .const import DOMAIN
+
+
+class HausbusDevice:
+    """Common base for Haus-Bus devices."""
+
+    def __init__(
+        self,
+        bridge_id: str,
+        device_id: str,
+        sw_version: str,
+        hw_version: str,
+        firmware_id: EFirmwareId,
+    ) -> None:
+        """Set up Haus-Bus device."""
+        self._device_id = device_id
+        self.manufacturer = "Haus-Bus.de"
+        self.model_id = "Controller"
+        self.name = f"Controller ID {self._device_id}"
+        self.software_version = sw_version
+        self.hardware_version = hw_version
+        self.bridge_id = bridge_id
+        self.firmware_id = firmware_id
+
+    @property
+    def device_id(self) -> str | None:
+        """Return a serial number for this device."""
+        return self._device_id
+
+    @property
+    def device_info(self) -> DeviceInfo | None:
+        """Return a device description for device registry."""
+        if self.device_id is None:
+            return None
+
+        return DeviceInfo(
+            identifiers={(DOMAIN, self.device_id)},
+            manufacturer=self.manufacturer,
+            model=self.model_id,
+            name=self.name,
+            sw_version=self.software_version,
+            hw_version=self.hardware_version,
+            via_device=(DOMAIN, self.bridge_id),
+        )
+
+    def set_type(self, type_id: int):
+        """Set device name and model_id according to device type."""
+        match self.firmware_id:
+            case EFirmwareId.ESP32:
+                match type_id:
+                    case 0x65:
+                        self.model_id = "LAN-RS485 Brückenmodul"
```

Comment:
Instead of using nested match cases I would use a dictionary like this:

```python
model_ids = {
  EFirmwareID.ESP32: {
    0.65: "LAN-RS485 Brückenmodul",
    ...
  },
  ...
}
``` 

And then use `get` to get the default value of "Controller" in this case:

```python
firmware_model_id = model_ids.get(self.firmware_id, "Controller")
self.model_id = firmware_id if type(firmware_model_id) is str else firmware_model_id.get(type_id, "Controller")
```

Reply:
Or indeed as zweckj said try to provide translations. You will probably still need a dictionary for that though